lookout/style/format/feature_extractor.py

```diff
@@ -742,33 +741,31 @@ def _parse_file(self, contents: str, root: bblfsh.Node, path: str) -> \
         node_tokens.append(sentinel)
 
         # scan `node_tokens` and fill the gaps with imaginary nodes
-        result = []
         pos = 0
         parser = self.tokens.PARSER
-        searchsorted = numpy.searchsorted
         for node in node_tokens:
             if node.start_position.offset < pos:
                 continue
             if node.start_position.offset > pos:
                 sumlen = 0
                 diff = contents[pos:node.start_position.offset]
                 for match in parser.finditer(diff):
-                    positions = []
+                    offsets = []
                     for suboff in (match.start(), match.end()):
-                        offset = pos + suboff
-                        line = searchsorted(line_offsets, offset, side="right")
-                        col = offset - line_offsets[line - 1] + 1
-                        positions.append(Position(offset, line, col))
+                        offsets.append(pos + suboff)
                     token = match.group()
                     sumlen += len(token)
-                    result.append(VirtualNode(token, *positions, path=path))
+                    file.add(RawTokenAnnotation(*offsets))
                 assert sumlen == node.start_position.offset - pos, \
                     "missed some imaginary tokens: \"%s\"" % diff
             if node is sentinel:
                 break
-            result.extend(VirtualNode.from_node(node, contents, path, self.token_unwrappers))
+            uast_node_annot = list(VirtualNode.from_node(node, contents, self.token_unwrappers))
+            file.update(uast_node_annot)
             pos = node.end_position.offset
-        return result, parents
+
+        # backward
```

Comment:
backward?

Reply:
yes, I call a function to convert a new format to old vnodes.

I'll elaborate the comment